docs/content/installation/upgrade-with-helm.md

```diff
@@ -0,0 +1,107 @@
+---
+title: Upgrading NGINX Ingress Controller with Helm
+description: "This document describes how to use Helm to upgrade NGINX Ingress Controller from 2.x and 3.0.x to the latest available release."
+weight: 1800
+doctypes: [""]
+toc: true
+docs: "DOCS-603"
+---
+
+## Background
+
+In NGINX Ingress Controller version 3.1.0, [changes were introduced](https://github.com/nginxinc/kubernetes-ingress/pull/3606) to Helm resource names, labels and annotations to fit with Helm best practices.
+When using Helm to upgrade from a version prior to to 3.1.0, certain resources like Deployment, DaemonSet and Service will be recreated due to the aforementioned changes, which will result in downtime.
+
+Although the advisory is to update all resources in accordance with new naming convention, to avoid the downtime please follow the steps listed in this page.
+
+## Upgrade Steps
+{{<note>}} The following steps apply to both 2.x and 3.0.x releases.{{</note>}}
+
+The steps you should follow depend on the Helm release name:
+
+{{<tabs name="upgrade-helm">}}
+
+{{%tab name="Helm release name is `nginx-ingress`"%}}
+
+1. Use `kubectl describe deployments` to get the `Selector` value:
+
+    ```shell
+    kubectl describe deployments -n <namespace>
+    ```
+    Copy the key=value under `Selector`, such as:
+
+    ```shell
+    Selector:               app=nginx-ingress-nginx-ingress
+    ```
+
+1. Checkout the latest available tag using `git checkout v3.3.0`
+
+1. Update the `selectorLabels: {}` field in the `values.yaml` file located at `/kubernates-ingress/deployments/helm-chart` with the copied `Selector` value.
+    ```shell
+    selectorLabels: {app: nginx-ingress-nginx-ingress}
+    ```
+
+1. Run `helm upgrade` with following arguments set:
+    ```shell
+    --set controller.serviceNameOverride="nginx-ingress-nginx-ingress"
+    --set controller.name=""
+    --set fullnameOverride="nginx-ingress-nginx-ingress"
+    ```
```

Comment:
Should we also provide an example of how to do a upgrade to point to a `values.yaml` file as well?
They can make the changes in their `values.yaml` file and upgrade with OCI method:
Something like:

```
helm upgrade <nic_release_name> oci://ghcr.io/nginxinc/charts/nginx-ingress--version 0.18 -n nginx-ingress  -f values.yaml
```
Thoughts? 🤔

Reply:
for OCI method, all the user has to do is `repository: "oci://ghcr.io/nginxinc/charts/nginx-ingress"` in values.yaml or `--set controller.repository="repository: "oci://ghcr.io/nginxinc/charts/nginx-ingress" ` which isn't upgrade specific and applies to both installation and upgrade.

Pointing to values is separate from OCI so can defiantly add it as modifying values.yaml is integral part for this.

Reply:
It's not NIC repository, it's the repository for the Helm Chart. It's a separate argument.

Reply:
gotcha I guess we don't need it then, user can use whatever desired to keep it more generalized